openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -429,9 +429,28 @@ def process_resources(self, instance, staging_dir):
                 # node doesn't have color space attribute
                 color_space = "Raw"
             else:
-                if files_metadata[source]["color_space"] == "Raw":
-                    # set color space to raw if we linearized it
-                    color_space = "Raw"
+                # get all the resolved files
+                metadata = files_metadata.get(source)
+                if metadata:
+                    metadata = files_metadata[source]
+                    if metadata["color_space"] == "Raw":
+                        # set color space to raw if we linearized it
+                        color_space = "Raw"
+                else:
+                    # if the files are unresolved from `source`
+                    # assume color space from the first file of 
+                    # the resource
+                    metadata = files_metadata.get(source)
+                    if not metadata:
+                        first_file = next(iter(resource.get(
+                            "files", [])), None)
+                        if not first_file:
+                            continue
+                    first_filepath = os.path.normpath(first_file)
+                    metadata = files_metadata[first_filepath]
+                    if metadata["color_space"] == "Raw":
+                        # set color space to raw if we linearized it
+                        color_space = "Raw"
```

Comment:
```suggestion
                # if the files are unresolved from `source`
                # assume color space from the first file of 
                # the resource
                metadata = files_metadata.get(source)
                if not metadata:
                    first_file = next(iter(resource.get(
                        "files", [])), None)
                    if not first_file:
                        continue
                first_filepath = os.path.normpath(first_file)
                metadata = files_metadata[first_filepath]
                if metadata["color_space"] == "Raw":
                    # set color space to raw if we linearized it
                    color_space = "Raw"
```
You seemed to have some duplicated code in there.

Reply:
It's also the duplicated usage of this:
```

Reply:
ah i see. will fix it as soon as possible